### PR TITLE
Break out interactive provider for more testing

### PR DESCRIPTION
**Pull request by phobologic**
This is in preparation using change sets for diffing.

### CODE REVIEW

File: stacker/providers/aws/interactive.py [hunk 1]
```diff
@@ -103,75 +108,123 @@ def output_summary(fqn, action, changeset, replacements_only=False):
...
+                                  sleep_time=.1):
+    """ Checks state of a changeset, returning when it is in a complete state.
 
-    def __init__(self, *args, **kwargs):
-        self.replacements_only = kwargs.pop('replacements_only', False)
-        super(Provider, self).__init__(*args, **kwargs)
+    Since changesets can take a little bit of time to get into a complete
+    state, we need to poll it until it does so. This will try to get the
+    state 5 times, waiting 2 seconds between each try. If, after that time,
+    the changeset is in a complete state it fails.
```
Comment (mmdriley) [inline stacker/providers/aws/interactive.py]: is *not* in a complete state

File: stacker/providers/aws/interactive.py [hunk 2]
```diff
@@ -103,75 +108,123 @@ def output_summary(fqn, action, changeset, replacements_only=False):
...
+def wait_till_change_set_complete(cfn_client, change_set_id, try_count=5,
+                                  sleep_time=.1):
+    """ Checks state of a changeset, returning when it is in a complete state.
 
-    def __init__(self, *args, **kwargs):
-        self.replacements_only = kwargs.pop('replacements_only', False)
-        super(Provider, self).__init__(*args, **kwargs)
+    Since changesets can take a little bit of time to get into a complete
+    state, we need to poll it until it does so. This will try to get the
+    state 5 times, waiting 2 seconds between each try. If, after that time,
```
Comment (mmdriley) [inline stacker/providers/aws/interactive.py]: default isn't 2 seconds.

**Comment by phobologic** [comment]
Thanks @mmdriley - updated!

**Comment by ttaub** [comment]
@phobologic are you going to merge this in?

**Pull request merged**